src/v/ssx/sugar.h

```diff
@@ -0,0 +1,36 @@
+/*
```

Comment:
I think this should be `ssx/semaphore.h` and include `ssx::named_mutex` (that doesn't yest exist) from `utils/mutex.h` (not sure there needs to be an unnamed version any more).